app/uk/gov/nationalarchives/omega/editorial/models/UpdateEditSetRecord.scala

```diff
@@ -22,11 +22,44 @@
 package uk.gov.nationalarchives.omega.editorial.models
 
 import play.api.libs.json.{ Format, Json }
+import uk.gov.nationalarchives.omega.editorial.models.UpdateEditSetRecord.Fields
 
-import java.time.LocalDateTime
+import java.time.{ LocalDate, LocalDateTime }
 
-case class UpdateEditSetRecord(editSetOci: String, recordOci: String, timestamp: LocalDateTime)
+case class UpdateEditSetRecord(editSetOci: String, recordOci: String, timestamp: LocalDateTime, fields: Fields)
 
 object UpdateEditSetRecord {
+
   implicit val format: Format[UpdateEditSetRecord] = Json.format[UpdateEditSetRecord]
+
+  case class Fields(
```

Comment:
I just want to make sure I understand the logic here:
1. The user wants to update a field, for example, `coveringDates`. They submit the form with the updated value.
2. When the form is submitted, it is populated with all the existing values from the model.
3. The values from the populated form is used to instantiate this class, `Fields`.
4. All fields of the model are updated, not just `coveringDates`

Is this correct?

Reply:
Yes, that's correct. Or at least, it reflects my understanding of the API.

> All fields of the model are updated, not just coveringDates

I wouldn't use the word "updated", as the value may not have changed from the original.